### PR TITLE
Harvard location

### LINKED ISSUE #13: Set up location controller and tests
Set up location controller and tests

## Acceptance criteria

- [ ] The location controller is fully implemented according to the instructions in <https://ucsb-cs156.github.io/f22/lab/team01/>
- [ ] Appropriate tests are in place for the location controller

### LINKED ISSUE #14: Set up location service and tests
Set up location service and tests

## Acceptance criteria

- [ ] The location service is fully implemented according to the instructions in <https://ucsb-cs156.github.io/f22/lab/team01/>
- [ ] Appropriate tests are in place for the location service

### PR DESCRIPTION
**Pull request by HarvardWang**
This PR satisfies tasks #13 and #14 

Finished Location updates

### CODE REVIEW

**Comment by Gopu2001** [comment]
Can you link the issues that are to be marked as done via this branch? i.e. 13 &/ 14

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/CountryCodeQueryService.java [hunk 1]
```diff
@@ -40,7 +40,7 @@ public String getJSON(String country) throws HttpClientErrorException {
 
         Map<String, String> uriVariables = Map.of("country", country);
 
-        HttpEntity<String> entity = new HttpEntity<>("body", headers);
+        HttpEntity<String> entity = new HttpEntity<>(headers);
```
Comment (Gopu2001) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/services/CountryCodeQueryService.java]: I don't think we should modify the CountryCodeQueryService file unless it fails some test

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryService.java [hunk 2]
```diff
@@ -16,9 +29,20 @@ public LocationQueryService(RestTemplateBuilder restTemplateBuilder) {
...
+        headers.setAccept(List.of(MediaType.APPLICATION_JSON));
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        Map<String, String> uriVariables = Map.of("location", location);
+
+        HttpEntity<String> entity = new HttpEntity<>("body", headers);
+
+        ResponseEntity<String> re = restTemplate.exchange(ENDPOINT, HttpMethod.GET, entity, String.class,
+        uriVariables);
+return re.getBody();
```
Comment (Gopu2001) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryService.java]: Can you tab this over for code readability improvement?

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryService.java [hunk 3]
```diff
@@ -16,9 +29,20 @@ public LocationQueryService(RestTemplateBuilder restTemplateBuilder) {
...
     }
 }
```
Comment (Gopu2001) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/services/LocationQueryService.java]: Can you add a \n to the end of every file so we can ignore this type of change in future git diffs?

File: src/test/java/edu/ucsb/cs156/spring/backenddemo/controllers/LocationControllerTests.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    MvcResult response = mockMvc
+        .perform( get(url).contentType("application/json"))
+        .andExpect(status().isOk()).andReturn();
+
+    String responseString = response.getResponse().getContentAsString();
+
+    assertEquals(fakeJsonResult, responseString);
+  }
+
+}
```
Comment (Gopu2001) [inline src/test/java/edu/ucsb/cs156/spring/backenddemo/controllers/LocationControllerTests.java]: fix this git diff warning

**Review by Gopu2001 — CHANGES_REQUESTED** [review]
overall, looks good. Slight suggestions and requested changes.

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/services/CountryCodeQueryService.java (hunk 1 above)
Comment (HarvardWang) [reply src/main/java/edu/ucsb/cs156/spring/backenddemo/services/CountryCodeQueryService.java]: Woops, yeah I think I edited this on accident

**Review by Gopu2001 — APPROVED** [review]
LGTM --> Looks good to me!

**Pull request merged**